examples/reinforcement_learning/dqn_cartpole.py

```diff
@@ -0,0 +1,187 @@
+        else:
+            raise ValueError('Unsupported target_type: {}'.format(target_type))
+    loss = mean_clipped_loss(y, t)
+    Q.cleargrads()
+    loss.backward()
+    opt.update()
+
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Chainer example: DRL(DQN)')
+    parser.add_argument('--env', type=str, default='CartPole-v0',
+                        help='Name of the OpenAI Gym environment to play')
+    parser.add_argument('--batchsize', '-b', type=int, default=64,
+                        help='Number of transitions in each mini-batch')
+    parser.add_argument('--episodes', '-e', type=int, default=1000,
+                        help='Number of episodes to run')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='dqn_result',
+                        help='Directory to output the result')
+    parser.add_argument('--unit', '-u', type=int, default=100,
+                        help='Number of units')
+    parser.add_argument('--target-type', type=str, default='dqn',
+                        help='Target type', choices=['dqn', 'double_dqn'])
+    parser.add_argument('--reward-scale', type=float, default=1e-2,
+                        help='Reward scale factor')
+    parser.add_argument('--replay-start-size', type=int, default=500,
+                        help='Number of steps after which replay is started')
+    parser.add_argument('--steps-to-decay-epsilon', type=int, default=5000,
+                        help='Number of steps used to linearly decay epsilon')
+    parser.add_argument('--min-epsilon', type=float, default=0.01,
+                        help='Minimum value of epsilon')
+    parser.add_argument('--target-update-freq', type=int, default=100,
+                        help='Frequency of target network update')
+    parser.add_argument('--record', action='store_true', default=True,
+                        help='Record performance')
+    parser.add_argument('--no-record', action='store_false', dest='record')
+    args = parser.parse_args()
+
+    # Initialize an environment
+    env = gym.make(args.env)
+    assert isinstance(env.observation_space, gym.spaces.Box)
+    assert isinstance(env.action_space, gym.spaces.Discrete)
+    ndim_obs = env.observation_space.low.size
+    n_actions = env.action_space.n
+    if args.record:
+        env.monitor.start(args.out, force=True)
+
+    # Initialize variables
+    D = collections.deque(maxlen=10 ** 6)
+    Rs = collections.deque(maxlen=100)
+    step = 0
+
+    # Initialize a model and its optimizer
+    Q = QFunction(ndim_obs, n_actions, n_units=args.unit)
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        Q.to_gpu(args.gpu)
+    target_Q = copy.deepcopy(Q)
+    opt = optimizers.Adam(eps=1e-2)
+    opt.setup(Q)
+
+    for episode in range(args.episodes):
```

Comment:
Since some envs defines the reward threshold at which they are considered solved, I added code to check it.

The size of D doesn't affect the time spent on each episode, so I think it doesn't matter.